File: Assets/ToluaFramework/Scripts/Utility/ModelManager/ModelManager.cs
using UnityEngine;

public class ModelManager 
{
    #region Instance

    /// <summary>
    /// 
    /// </summary>
    private static ModelManager mInstance = new ModelManager();
 
    /// <summary>
    /// 
    /// </summary>
    public static ModelManager instance
    {
        get { return mInstance; }
    }

    #endregion

    #region Public

    /// <summary>
    /// 
    /// </summary>
    public void Setup()
    {

    }

    /// <summary>
    /// 
    /// </summary>
    /// <param name="assetPath"></param>
    /// <param name="assetName"></param>
    /// <returns></returns>
    public GameObject Load(string assetPath, string assetName)
    {
        return AssetPoolManager.instance.Alloc(AssetPoolManager.Type.Model, assetPath, assetName) as GameObject;
    }

    /// <summary>
    /// 
    /// </summary>
    /// <param name="model"></param>
    public void Unload(GameObject model)
    {
        AssetPoolManager.instance.Dealloc(AssetPoolManager.Type.Model, model);
    }

    #endregion
}

File: Assets/ToluaFramework/Lua/manager/modelManager.lua
--region *.lua
--Date
--此文件由[BabeLua]插件自动生成

local modelManager = { }
assetType = "model"

-------------------------------------------------------------------
-- 全局初始化
-------------------------------------------------------------------
function modelManager.setup()
    AssetPoolManager.instance:AddPool(assetType, "model", false)
end

-------------------------------------------------------------------
-- 加载资源
-------------------------------------------------------------------
function modelManager.load(assetPath, assetName)
    return AssetPoolManager.instance:Alloc(assetType, string.lower(assetPath), string.lower(assetName))
end

-------------------------------------------------------------------
-- 卸载资源
-------------------------------------------------------------------
function modelManager.unload(model)
    AssetPoolManager.instance:Dealloc(assetType, model.gameObject)
end

return modelManager

--endregion
